User/chigy/roundedcorner (#34)
microsoft/microsoft-ui-xaml-specs: active/RoundedCorner/roundedcorner.md

![bars](https://github.com/microsoft/microsoft-ui-xaml-specs/blob/user/chigy/roundedcorner/active/RoundedCorner/ImageFiles/Bars.png)

**Controls being rounded**
- NavigationView
- Pivot
- ScrollIndicator
- ProgressBar
- Slider
- ColorPicker
- MediaPlayerElement
- WebView (may or may not get it from XAML change, checking)

*When not to round*
- When a multiple buttons are inside a single button that houses them (e.g. SplitButton).
![SplitButton](https://github.com/microsoft/microsoft-ui-xaml-specs/blob/user/chigy/roundedcorner/active/RoundedCorner/ImageFiles/SplitButton2.png)
- When a control is housed inside another container (e.g. ScrollBar's bar and buttons that are part of the ScrollBar container which is also part of a ScrollViewer).

![ScrollBar](https://github.com/microsoft/microsoft-ui-xaml-specs/blob/user/chigy/roundedcorner/active/RoundedCorner/ImageFiles/ScrollBar.png)
- When flyout UI element that is connected to a UI that invokes the flyout on one side.
![AutoSuggest](https://github.com/microsoft/microsoft-ui-xaml-specs/blob/user/chigy/roundedcorner/active/RoundedCorner/ImageFiles/AutoSuggest.png)

> ## Note on keyboard focus rect and shadow

Reviewer: Would it be good to also show some screenshots of what it looks like if you round too much with a control that has shadows?

Author: @kikisaints , good idea! Would you take a screenshot of something to add here since you are set up for that? It doesn't have to be final one we use for the official doc. Just a concept to show it visually inline to make sure it is a good idea. Let me know if you are not easily set up for that.

Reviewer: Sure! Let me know how these look: 
![ThemeShadow_ExtremeRoundedCornersExample](https://user-images.githubusercontent.com/6964369/60532329-ee241f80-9cb1-11e9-9268-67036d840eff.png)

Left one is 4px rounding, right is 15.

Reviewer 2: @kikisaints if you want to see more range the WinUI test app has a slider where you can play around with different CornerRadius values across all controls. There are some issues with some of the flyouts (because you can't x:Bind to a style value), but ContentDialog works.

Author: I've incorporated the image (so I think...) into the doc.

> Currently our default design does not require any special work to round the corners of keyboard focus rectangle and shadow. Using higher corner radius value will not break them functionally, however it is good to be aware of this to avoid unwanted visual glitches that you could introduce with a larger value.

Author: ```suggestion
Currently our default design does not require any special work to round the corners of keyboard focus rectangle and shadow. Using higher corner radius value will not break them functionally, however it is good to be aware of this to avoid unwanted visual glitches that you could introduce with a larger value.

Here is an example of larger corner radius make the shadow look undesirable.
![ContentDialogShadow](https://user-images.githubusercontent.com/6964369/60532329-ee241f80-9cb1-11e9-9268-67036d840eff.png)
```

*Corners of rectangle UI elements*
- These UI elements include basic UIs like buttons that users see them on a surface of the screen at all times.
- The default radius value we use are 2px for these UI.
![Button](https://github.com/microsoft/microsoft-ui-xaml-specs/blob/user/chigy/roundedcorner/active/RoundedCorner/ImageFiles/Button.png)

**Controls being rounded**
- Button
- CheckBox
- ComboBox
- DropDownButton
- Slider
- SplitButton
- ToggleButton
- ToggleSplitButton
- Flipview
- GridView
- ListView
- TreeView
- ContentDialog
- AutoSuggestBox
- PasswordBox
- RichEditBox
- TextBox
- DatePicker
- CalendarDatePicker
- Tab control

*Corners of flyout UI elements*
- These UI elements are those that appear temporarily on UI like a MenuFlyout or Flyout.
- The default radius value we use here are 4px.
![flyout](https://github.com/microsoft/microsoft-ui-xaml-specs/blob/user/chigy/roundedcorner/active/RoundedCorner/ImageFiles/Flyout.png)

**Controls being rounded**
- CalendarDatePicker
- DatePicker
- TimePicker
- Flyout
- TeachingTip
- ToolTip
- DropDownButton
- SplitButton

> - Slider

Reviewer 3: This one is confusing, slider isn't a transient control. Are you talking about Slider's value tooltip? It may be worthwhile to call out which control part is being rounded, or at least say that this 4px radius rule doesn't necessary apply to the entire control listed.

Author: ```suggestion
- Slider (tooltip flyout only)
```

Author: @YuliKl , does this make clearler?

- AutoSuggestBox
- CommandBarFlyout
- MenuFlyout
- ComboBox
- ColorPicker
- MediaPlayerElement
- ContentDialog
- MenuBar
- ToggleSplitButton

*Rounding of bars*

> - These UI elements are bar like or line like UI (e.g. ProgressBar).

Reviewer 3: ```suggestion
- These UI elements are shaped like bars or lines, for example ProgressBar.
```

> - The default radius value we use here are 2px except for slider in ColorPicker which will be 6px because the bar is wider.

Reviewer 3: ```suggestion
- The default radius values we use here are 2px. One exception is the slider in ColorPicker which will be 6px because the bar is wider.
```

![bars](https://github.com/microsoft/microsoft-ui-xaml-specs/blob/user/chigy/roundedcorner/active/RoundedCorner/ImageFiles/Bars.png)

**Controls being rounded**

> - NavigationView

Reviewer 3: ```suggestion
- NavigationView's selection indicator
```

> - Pivot

Reviewer 3: ```suggestion
- Pivot's selection indicator
```

- ScrollIndicator
- ProgressBar
- Slider

> - ColorPicker

Reviewer 3: ```suggestion
- ColorPicker's slider
```

> - MediaPlayerElement

Reviewer 3: It's progress bar, right? Not the entire control?

Author: ```suggestion
- MediaPlayerElement (tooltip flyout only)
```

> - WebView (may or may not get it from XAML change, checking)

Reviewer 3: WebView's scroll indicator?

Author: ```suggestion
Note: WebView uses the scroll indicator coming from HTML directly. So this will not be rounded until scroll indicator from hosted HTML gets the rounded treatment.
```

> *When not to round*

Reviewer 3: ```suggestion
### When not to round
```

> - When a multiple buttons are inside a single button that houses them (e.g. SplitButton).

Reviewer 3: ```suggestion
- When multiple buttons are inside a single control that houses them (e.g. SplitButton).
```

![SplitButton](https://github.com/microsoft/microsoft-ui-xaml-specs/blob/user/chigy/roundedcorner/active/RoundedCorner/ImageFiles/SplitButton2.png)

> - When a control is housed inside another container (e.g. ScrollBar's bar and buttons that are part of the ScrollBar container which is also part of a ScrollViewer).

Reviewer 3: I don't quite buy the logic of this one, especially the mention of ScrollViewer. The scroll indicator is also inside the same ScrollViewer, but it is getting rounded. Perhaps focusing on just the scroll bar container would be better. But that logic doesn't hold up for progress bar.

Reviewer 4: Perhaps wording it more like:
**When a control is touching the edges of the window frame or screen, it should not be rounded**

That would cover the ApplicationBar and the ScrollViewer, but does not account for when the ScrollViewer is used for a component within a page, but not flush with the edge of the window.

Author: The way @mdtauk put it eliminates the DropDownButton so cannot do that...

Author: How about...
"When multiple UI elements touches each other that is housed inside a container (e.g. ScrollBar's bar and buttons touch each other inside the ScrollBar container), there should be no space when they contact."

Author: @YuliKl  and @mdtauk , since I didn't hear otherwise, I've committed the change I suggested, but you are welcome and encouraged to provide feedback if this is still NG.

**Controls being rounded**
- CalendarDatePicker
- DatePicker
- TimePicker
- Flyout
- TeachingTip
- ToolTip
- DropDownButton
- SplitButton
- Slider (tooltip flyout only)
- AutoSuggestBox
- CommandBarFlyout
- MenuFlyout
- ComboBox
- ColorPicker
- MediaPlayerElement
- ContentDialog
- MenuBar
- ToggleSplitButton

*Rounding of bars*
- These UI elements are shaped like bars or lines, for example ProgressBar.
- The default radius values we use here are 2px. One exception is the slider in ColorPicker which will be 6px because the bar is wider.
![bars](https://github.com/microsoft/microsoft-ui-xaml-specs/blob/user/chigy/roundedcorner/active/RoundedCorner/ImageFiles/Bars.png)

**Controls being rounded**
- NavigationView's selection indicator
- Pivot's selection indicator
- ScrollIndicator
- ProgressBar
- Slider
- ColorPicker's slider
- MediaPlayerElement (tooltip flyout only)

Note: WebView uses the scroll indicator coming from HTML directly. So this will not be rounded until scroll indicator from hosted HTML gets the rounded treatment.

### When not to round

> - When multiple buttons are inside a single control that houses them (e.g. SplitButton).

Author: ```suggestion
- When multiple UI elements touch each other that is housed inside a container (e.g. ScrollBar's bar and buttons touch each other inside the ScrollBar container), there should be no space when they contact.
```
